sound/soc/sof/intel/hda-codec.c

```diff
@@ -172,25 +173,26 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 		 * handle ret==0 (no driver bound) as an error, but pass
 		 * other return codes without modification
 		 */
-		if (ret == 0)
+		if (ret == 0) {
+			ret = -ENOENT;
 			goto error;
```

Comment:
Hmm, looking at this now, maybe I can remove got "goto error", just set 'ret' and let if fall through to error handling.